### DISCUSSION TITLE
Realistic timeline for semi-production ready build?

### DISCUSSION THREAD

**Discussion by ethixkr** [Textualize/textual#34] [Q&A]
Hey Will, just wanted to begin with a thank you from the bottom of my heart. The heart and soul you pour into your work is evident by the quality projects you put out, from Rich to pyfilesystem. Huge fan of your work.

With that being said, I don't mean to rush you, but I'm just so in love with Rich and it seems like you're making great progress on transforming Rich into a whole new beast. 

I know you've said in an earlier post that it'd we'd be better off continuing to build on Rich and port it over relatively easily to Textual when it's more ready. From my limited perspective, it looks pretty darn ready already besides the lack of Windows support and proper documentation.  😎 

I also know that being (for the most part) the sole dev for a project can be exhausting and you have a life too! Hahah, like I said, I don't want to rush you or anything (😉) but what's the realistic timeline looking like? Will I regret it dearly if I jump the gun and start building our next project despite your warnings? Surely it can't be that much worse than trying to hack together interactivity on a Rich layout at the moment? Luckily 90% of what we're building will be on Linux-based systems.

Thoughts, comments, suggestions, as always, very much appreciated -- especially from you.

P.S. I would absolutely donate to the project, just can't seem to find the sponsor button ;p

**Comment by willmcgugan** [comment]
Gad you like my work!

If you are hacking interactivity on top of Rich then you may be better off starting with Textual. But you would be a very early adopter and there are downsides to that. The API isn't stable yet, so things may (and likely will) change from one release to the next. Documentation won't be ready for a while and I can't guarantee if you find a blocking bug that I will fix it quickly.

I also can't guarantee any timescale for a v1.0 release. Realistically I'd guess 3-6 months.

If you do want to go ahead with Textual, I'd suggest looking at the examples. If you need anything that is not in the examples them you would need to implement that yourself, or wait for me. And pin the Textual version to the minor version number so a new release won't break your app.

See the [Rich Repo](https://github.com/willmcgugan/rich) if you would like to Sponsor :)

**Reply by ethixkr** [reply]
Definitely will donate from time to time :) Cheers!

**Comment by Njordy** [comment]
Any news on how is it going, and when a Windows support might be added? :)
Thanks
